src/annoymodule.cc

```diff
@@ -237,16 +237,25 @@ py_an_save(py_annoy *self, PyObject *args, PyObject *kwargs) {
 PyObject*
 get_nns_to_python(const vector<int32_t>& result, const vector<float>& distances, int include_distances) {
   PyObject* l = PyList_New(result.size());
+  if (l == NULL) {
+    return NULL;
+  }
   for (size_t i = 0; i < result.size(); i++)
     PyList_SetItem(l, i, PyInt_FromLong(result[i]));
   if (!include_distances)
     return l;
 
   PyObject* d = PyList_New(distances.size());
+  if (d == NULL) {
+    return NULL;
```

Comment:
This leaks a reference to `l`. The standard approach is to have a `goto error`, and the `error` block `Py_XDECREF`s all the things that may have been allocated.

Reply:
I've opened a PR to address your feedback. if you don't mind it would be great to have your review

https://github.com/spotify/annoy/pull/552